linetools/spectra/xspectrum1d.py

```diff
@@ -231,7 +239,8 @@ def copy(self):
     def wavelength(self):
         """ Return the wavelength array with units
         """
-        return self.data[self.select]['wave'] * self.units['wave']
+        #return self.data[self.select]['wave'] * self.units['wave']
+        return self.data['wave'][self.select].compressed() * self.units['wave']
```

Comment:
Not sure about compressed. If the spectrum has gaps, not sure how this compressed data would look like.